### PR TITLE
Add bors-ng configuration

### LINKED ISSUE #491: Set up merge bot for repository
You know what's annoying? Having to manually rebase every PR, often multiple times, before it be can merged.

As suggested by @cryslith, let's set up [Bors-NG](https://bors.tech/). This will allow us to adjust our workflow to stop being based in rebasing everything -- which is somewhat easy for new contributors to get wrong, beyond being a waste of valuable developer time.

### PR DESCRIPTION
**Pull request by cryslith**
- Configure bors-ng
- Run CircleCI only during bors merges
- Require branches to have linear git history

Closes #491.

---

Checklist:

 - [x] I have split up this change into one or more appropriately-delineated commits.
 - [x] The first line of each commit is of the form "[component]: do something"
 - [x] I have written a complete, multi-line commit message for each commit.
 - [x] I have formatted any Go code that I have changed with gofmt.
 - [x] I have written or updated appropriate documentation to cover this change.
 - [x] I have confirmed that this change is covered by at least one appropriate test run by CI.
 - [x] If my change includes new or modified functionality, I have tested that the changes work as expected.
 - [x] I have assigned this issue to an appropriate reviewer. (Choose @celskeggs if you are not otherwise certain.)
 - [ ] I consider my PR complete and ready to be merged without my further input, assuming that it passes CI and code review.
 - [x] My changes have passed CI, including an automatic Jenkins deploy.
 - [x] My changes have passed code review.

### CODE REVIEW

**Comment by cryslith** [comment]
Note: for this to work, I think we'll need to change the repo configuration to require `continuous-integration/jenkins/branch` instead of `continuous-integration/jenkins/pr-merge`.  We would also need to replace the "Require pull request reviews before merging" option with requiring the `bors` check to pass.

File: .circleci/config.yml [hunk 1]
```diff
@@ -32,3 +32,14 @@ jobs:
...
+  version: 2
+  build:
+    jobs:
+      - build:
+          filters:
+            branches:
+              only:
+                - staging
+                - trying
+                - master
```
Comment (celskeggs) [inline .circleci/config.yml:60]: Do we want to apply this same configuration to Jenkins?

Comment (cryslith) [reply .circleci/config.yml:60]: I'm not sure how to do that - I think it requires configuring jenkins through the web interface?

**Comment by cryslith** [comment]
bors try

**Comment by cryslith** [comment]
bors try

File: .circleci/config.yml (hunk 1 above)
Comment (cryslith) [reply .circleci/config.yml:60]: This is now working; it's configured in jenkins itself.

File: docs/bors-ng-setup.txt [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+
+# use this command to generate random secrets when called for
+</dev/urandom tr -dc 'a-zA-Z0-9' | head -c64
+
+useradd -m -U borsng
+
+sudo -u postgres createuser -P borsng # generate random password
+sudo -u postgres createdb -O borsng borsng
+sudo -u postgres psql -d borsng <<<"CREATE EXTENSION IF NOT EXISTS citext;"
```
Comment (celskeggs) [inline docs/bors-ng-setup.txt:31]: Just to confirm -- postgres creates a localhost-only server, right? There's no chance of someone else being able to connect?

(I know we have a password, but as a second layer of precaution.)

Comment (cryslith) [reply docs/bors-ng-setup.txt:31]: Yep, default configuration is 

```
#listen_addresses = 'localhost'         # what IP address(es) to listen on;
                                        # comma-separated list of addresses;
                                        # defaults to 'localhost'; use '*' for all
```

File: docs/bors-ng-setup.txt [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+cat >~/bors-env <<EOF
+PORT=4001
+MIX_ENV=prod
+SECRET_KEY_BASE=???  # generate this randomly
+DATABASE_URL='ecto://borsng:<db password>@localhost/borsng'  # password from earlier
+GITHUB_INTEGRATION_ID=???  # App id in github
+GITHUB_WEBHOOK_SECRET=???  # generate this randomly and input it to github
+GITHUB_CLIENT_ID=???  # from github
+GITHUB_CLIENT_SECRET=???  # from github
+PUBLIC_HOST=localhost
```
[the remark this answers is not in the document]
Comment (celskeggs) [reply docs/bors-ng-setup.txt:111]: You mean `hijinks.mit.edu:4002`?

**Comment by cryslith** [comment]
yesOn Apr 20, 2020 11:11, Cel Skeggs <notifications@github.com> wrote:
@celskeggs commented on this pull request.



In docs/bors-ng-setup.txt:
> +MIX_ENV=prod mix phx.digest
+
+MIX_ENV=prod mix compile
+MIX_ENV=prod mix release
+
+
+cat >~/bors-env <<EOF
+PORT=4001
+MIX_ENV=prod
+SECRET_KEY_BASE=???  # generate this randomly
+DATABASE_URL='ecto://borsng:<db password>@localhost/borsng'  # password from earlier
+GITHUB_INTEGRATION_ID=???  # App id in github
+GITHUB_WEBHOOK_SECRET=???  # generate this randomly and input it to github
+GITHUB_CLIENT_ID=???  # from github
+GITHUB_CLIENT_SECRET=???  # from github
+PUBLIC_HOST=localhost

You mean hijinks.mit.edu:4002?

File: docs/bors-ng-setup.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (cryslith) [reply docs/bors-ng-setup.txt:111]: yes

**Comment by cryslith** [comment]
Reset

**Pull request merged**